### PR TITLE
Add planning options inside circle_motion.py

### PR DESCRIPTION
**Pull request by KahnShi**
Add motion planning options inside circle_motion.py.
Add environment for collision test.

### CODE REVIEW

**Comment by tongtybj** [comment]
@KahnShi 

Why are there two same  PR? #31 and #30

**Comment by KahnShi** [comment]
@tongtybj One for master branch, one for planning branch. And I am also thinking about whether we need planning branch here.

**Comment by tongtybj** [comment]
common sense: 
PR to the main branch, that is master in this repo.

**Comment by KahnShi** [comment]
Thank you for your explanation.
I removed.

**Comment by chibi314** [comment]
Functions isConvergent, saturateVelocity, goPos and goCircle are copy-and-pasted. So we should consider better design of files.
I suggest creating 'drone_navigation.py'. It has four functions noted above and related params.

**Comment by KahnShi** [comment]
I agree with you. @chibi314 
By the way, when in the state of [approaching to target](https://github.com/tongtybj/jsk_uav_forest/blob/master/jsk_uav_forest_motion/script/circle_motion.py#L237), do you think it needs to be revised as the following to be more accurate?
` 
tree_direction_ang = math.atan2(self.tree_xy_pos_[1], self.tree_xy_pos_[0])

vel_msg = self.goPos(self.LOCAL_FRAME_, np.array([self.tree_xy_pos_[0] - self.circle_radius_ * math.cos(tree_direction_ang), self.tree_xy_pos_[1]] - self.circle_radius_ * math.sin(tree_direction_ang)), self.takeoff_height_, self.uav_yaw_ + tree_direction_ang)`

**Comment by chibi314** [comment]
@KahnShi 
>I agree with you. @chibi314
Could you implement and test that file?

>By the way, when in the state of approaching to target, do you think it needs to be revised as the following to be more accurate, in case when tree_direction_ang is not so close to 0?
I think it is better, so please revise it. Thanks.

**Comment by KahnShi** [comment]
@chibi314 
You are welcome, I have implemented and tested it, and make a PR #36 here.

**Comment by tongtybj** [comment]
Check #37 which discuss the related issue. I don't want to stop this PR, just want everyone to recognized some potential problem will occur.

**Comment by tongtybj** [comment]
I think the obstacle avoidance mode should be the default function.
Simple suggestion is to combine the circle_motion.py and motion_plannar_client.py, or create the navigation.py separating control and navigation part.

@KahnShi @chibi314 
Let's have a small meeting tonight after me and shi coming back from kashiwa.

**Comment by tongtybj** [comment]
Resolve the ghost obstacle problem by #47

**Comment by tongtybj** [comment]
@KahnShi 

Remove the state machine like "init_state" and "safty_flying_state" in the planning node.
Change to a single flag which start the planning mode

**Comment by tongtybj** [comment]
@KahnShi 

Please add your message here.

**Comment by tongtybj** [comment]
@KahnShi 
[d24e170](https://github.com/tongtybj/jsk_uav_forest/pull/31/commits/d24e170862be5339d059b14bfa6246da4e1c99f1)和[b3423ed](https://github.com/tongtybj/jsk_uav_forest/pull/31/commits/b3423eda6599ccd9157e02d0263a0c406a6ab9c9)在做同样的事。

后者是我昨天push到shi/planning的，人多的话就不能这么开发了。这次我刻意这么做，是想让你在rebase时确认下前面的人都改了那些地方。

File: jsk_uav_forest_motion/script/motion_plannar_client.py [hunk 1]
```diff
@@ -150,15 +151,19 @@ def nearestObstacleSafetyDetection(self):
 
                 if abs(self.tree_cluster_.ranges[i]) < self.drone_obstacle_ignore_maximum_radius_:
```
Comment (tongtybj) [inline jsk_uav_forest_motion/script/motion_plannar_client.py]: @KahnShi 

I think
```
self.drone_obstacle_ignore_maximum_radius_
```
and 
```
nearest_obstacle_distance_in_head_direction
```
are doing the same thing.

Choose one of them.

Comment (tongtybj) [inline jsk_uav_forest_motion/script/motion_plannar_client.py]: @KahnShi 

Oh, sorry, I realized the necessity of both two variables.

**Pull request merged**